pytorch_lightning/plugins/training_type/ddp2.py

```diff
@@ -11,65 +11,15 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import torch
-
-from pytorch_lightning.plugins.training_type.ddp import DDPPlugin
-from pytorch_lightning.utilities.apply_func import apply_to_collection
-from pytorch_lightning.utilities.enums import _StrategyType
-from pytorch_lightning.utilities.types import _METRIC_COLLECTION
-
-
-class DDP2Plugin(DDPPlugin):
-    """DDP2 behaves like DP in one node, but synchronization across nodes behaves like in DDP."""
-
-    distributed_backend = _StrategyType.DDP2
-
-    @property
-    def global_rank(self) -> int:
-        return self.node_rank
-
-    @property
-    def world_size(self) -> int:
-        return self.num_nodes
-
-    def reduce(self, collection: _METRIC_COLLECTION, *args, **kwargs) -> _METRIC_COLLECTION:
-        """Reduces a collection of tensors from all processes. It can be applied to just a single tensor. In DDP2,
-        the reduction here is only across local devices within the node.
-
-        Args:
-            collection: The collection of tensors to sync and reduce.
-            *args: ignored for DDP2
-            **kwargs: ignored for DDP2
-
-        Return:
-            Reduced tensor values or the same value if it was not or did not contain a tensor.
-        """
-
-        def mean(t: torch.Tensor) -> torch.Tensor:
-            original_dtype = t.dtype
-            return t.float().mean().to(original_dtype)
-
-        return apply_to_collection(collection, torch.Tensor, mean)
-
-    @property
-    def root_device(self):
-        return self.parallel_devices[0]
-
-    def model_to_device(self):
-        # no need to do anything when model is wrapped in torch.nn.DataParallel
-        pass
-
-    @property
-    def distributed_sampler_kwargs(self):
-        distributed_sampler_kwargs = dict(num_replicas=self.num_nodes, rank=self.global_rank)
-        return distributed_sampler_kwargs
-
-    @property
-    def _is_single_process_single_device(self) -> bool:
-        return False
-
-    def set_world_ranks(self) -> None:
-        if self.cluster_environment is None:
-            return
-        self.cluster_environment.set_global_rank(self.node_rank)
-        self.cluster_environment.set_world_size(self.num_nodes)
+from pytorch_lightning.plugins.strategy.ddp2 import DDP2Plugin
+from pytorch_lightning.utilities import rank_zero_deprecation
+
+
+class DDP2Plugin(DDP2Plugin):
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        rank_zero_deprecation(
+            "`pytorch_lightning.plugins.training_type.ddp2.DDP2Plugin` has been deprecated in v1.6"
+            " and will be removed in v1.8. Use `pytorch_lightning.plugins.strategy.ddp2.DDP2Plugin`"
+            " instead."
+        )
```

Comment:
Since you are making these changes to each file, git does not properly move the file and interprets it as a deletion + addition.

I strongly suggest we do this in two PRs:

1. Move the code.
2. Add the backwards compatible paths.

This is to preserve the git history.

In between, the backwards compatibility would be broken but should be okay as there will not be a release in-between.

Since the Meta team relies on master, we just need them to skip the commits between step (1) until step (2) is merged. cc @ananthsub

Reply:
And move files lost the historical tracks, it's hard to tracking back changes and old issues related to it. Is this nessesary?

Reply:
`git mv` supports moving the location and changing the filename.

The problem is then adding changes to the file.

Reply:
> `git mv` supports moving the location and changing the filename.
> 
> The problem is then adding changes to the file.

Got you! that's why you suggested separate the steps, make sense! thanks~~